packages/mdc-typography/_mixins.scss

```diff
@@ -20,53 +20,78 @@
 // THE SOFTWARE.
 //
 
+@import "@material/feature-targeting/functions";
+@import "@material/feature-targeting/mixins";
 @import "./variables";
 
-@mixin mdc-typography-base {
-  @each $key, $value in $mdc-typography-base {
-    #{$key}: $value;
+@mixin mdc-typography-base($query: mdc-feature-all()) {
+  $feat-typography: mdc-feature-create-target($query, typography);
+
+  @include mdc-feature-targets($feat-typography) {
+    @each $key, $value in $mdc-typography-base {
+      #{$key}: $value;
+    }
   }
 }
 
-@mixin mdc-typography($style) {
+@mixin mdc-typography($style, $query: mdc-feature-all()) {
+  $feat-typography: mdc-feature-create-target($query, typography);
   $style-props: map-get($mdc-typography-styles, $style);
 
   @if not map-has-key($mdc-typography-styles, $style) {
     @error "Invalid style specified! #{$style} doesn't exist. Choose one of #{map-keys($mdc-typography-styles)}";
   }
 
-  @each $key, $value in $style-props {
-    #{$key}: $value;
+  @include mdc-feature-targets($feat-typography) {
+    @each $key, $value in $style-props {
+      #{$key}: $value;
+    }
   }
 }
 
 // Element must be `display: block` or `display: inline-block` for this to work.
-@mixin mdc-typography-overflow-ellipsis {
-  text-overflow: ellipsis;
-  white-space: nowrap;
-  overflow: hidden;
+@mixin mdc-typography-overflow-ellipsis($query: mdc-feature-all()) {
+  $feat-structure: mdc-feature-create-target($query, structure);
+
+  @include mdc-feature-targets($feat-structure) {
+    text-overflow: ellipsis;
+    white-space: nowrap;
+    overflow: hidden;
+  }
 }
 
-@mixin mdc-typography-baseline-top($distance) {
-  display: block;
-  margin-top: 0;
-  /* @alternate */
-  line-height: normal;
+@mixin mdc-typography-baseline-top($distance, $query: mdc-feature-all()) {
+  $feat-structure: mdc-feature-create-target($query, structure);
+
+  @include mdc-feature-targets($feat-structure) {
```

Comment:
why not include all the styles inside single `@include` block?

Reply:
I remember @kfranqueiro mentioning on another PR (can't remember where exactly) that he preferred this:

```scss
.some-selector {
  @include mdc-feature-targets($feat-structure) {
    ...
  }
}

.some-other-selector {
  @include mdc-feature-targets($feat-structure) {
    ...
  }
}
```
over this:

```scss
@include mdc-feature-targets($feat-structure) {
  .some-selector {
    ...
  }

  .some-other-selector {
    ...
  }
} 
```

I agree that this makes it easier to read the file because you can follow the selectors down first, and then worry about what feature you're targetting at the very end

Reply:
Ah ok. Thanks for explaining!

Yes, packaging feature related styles (ie, `structure`) into one block might be good for readability. But if you think it is consistent with other files this looks good.

WDYT @kfranqueiro ?

Reply:
Yeah, this was based on a comment I made on one of the other PRs. IIRC I specifically commented in a case where writing selectors nested within feature-targets blocks ended up causing redundant selectors in output compared to before, so this also serves to discourage that from happening, as well as just more closely mirroring how you'd write CSS if you were to ignore the feature-targets stuff.